impala/_thrift_api.py

```diff
@@ -345,24 +386,19 @@ def get_http_transport(host, port, http_path, timeout=None, use_ssl=False,
             auth = base64.encodebytes(user_password).decode().strip('\n')
         except AttributeError:
             auth = base64.encodestring(user_password).decode().strip('\n')
-
-        transport.setCustomHeaders({'Authorization': 'Basic %s' % auth})
+        transport.setDefaultCustomHeaders({'Authorization': 'Basic %s' % auth})
+        transport.setUseDynamicCustomHeaders()
 
     elif auth_mechanism == 'GSSAPI':
         # For GSSAPI over http we need to dynamically generate custom request headers.
-        def get_auth_headers(auth_cookie):
-            import kerberos
-            if auth_cookie:
-                cookie_value = auth_cookie.output(attrs=['value'], header='' ).strip()
-                return {'Cookie': cookie_value}
-            else:
-                _, krb_context = kerberos.authGSSClientInit("%s@%s" %
-                                    (kerberos_service_name, kerberos_host))
-                kerberos.authGSSClientStep(krb_context, "")
-                negotiate_details = kerberos.authGSSClientResponse(krb_context)
-                return {"Authorization": "Negotiate " + negotiate_details}
-
-        transport.setGetCustomHeadersFunc(get_auth_headers)
+        import kerberos
```

Comment:
Here, generating Authorization header in a callback was intentional. We shouldn't expect that the Authorization header generated at the beginning will be valid through the entire session. We might have to regenerate the header in the middle of a session.

Reply:
Done